### DISCUSSION TITLE
compact,delete Topic doenst get cleanup

### DISCUSSION THREAD

**Discussion by teimyBr** [strimzi/strimzi-kafka-operator#6792] [Q&A] [answered]
Hi i have a compact,delete Topic that doenst get cleanup after month. I have records in that topic that are older than 3 month. 
I think the with this settings the messages should be not older than 2 days. 
Strimzi 0.28.0
Broker Settings: 
    logMessageFormatVersion: "3.1"
    interBrokerProtocolVersion: "3.1"
    replicas: 5
    offsetsTopicReplicationFactor: 3
    transactionStateLogReplicationFactor: 3
    transactionStateLogMinInServiceReplicas: 2
    autoCreateTopicsEnable: false

Topic config:
        cleanup.policy: "compact,delete"
        retention.bytes: 104857600 # 100 MB (per partition)
        retention.ms: 172800000 # 2 days
        segment.ms: 60000 # 1 minute

Have somebody any idea ?

**Comment by sknot-rh** [comment]
Hi. There is also an option `log.retention.check.interval.ms`. After this period, the log is checked for "expired" records.
You can read more at https://strimzi.io/blog/2021/12/17/kafka-segment-retention/

**Comment by NickSalter** [comment]
Hi @teimyBr did you ever get an answer to this? We are experiencing the same issue.

**Reply by teimyBr** [reply, marked as answer]
@NickSalter  we resolved this by seeting only one of this two. 

retention.bytes: 104857600 # 100 MB (per partition)
retention.ms: 172800000 # 2 days

But i dont know why together it doesnt work. 

I read this is an OR Condition but i think this is AND. And then the Kafka Doku is wrong. 

So set only one of this two and it work
